src/mono/wasm/Wasm.Build.Tests/Templates/WasmTemplateTests.cs

```diff
@@ -627,5 +627,90 @@ void AssertFile(string suffix)
                 Assert.True(copyOutputSymbolsToPublishDirectory == File.Exists(Path.Combine(publishFrameworkPath, fileName)), $"The {fileName} file {(copyOutputSymbolsToPublishDirectory ? "should" : "shouldn't")} exist in publish folder");
             }
         }
+
+        [Theory]
+        [InlineData("Debug")]
+        [InlineData("Release")]
+        public async Task UCOWithSpecialCharacters(string config)
+        {
+            string code =
+                """
+                using System;
+                using System.Runtime.InteropServices;
+                public unsafe partial class Test
+                {
+
+                    public unsafe static int Main(string[] args)
+                    {
+                        ((IntPtr)(delegate* unmanaged<int,int>)&Interop.ManagedFunc).ToString();
+
+                        Console.WriteLine($"main: {args.Length}");
+                        Interop.UnmanagedFunc();
+                        return 0;
+                    }
+                }
+
+                file partial class Interop
+                {
+                    [UnmanagedCallersOnly(EntryPoint = "ManagedFunc")]
+                    public static int ManagedFunc(int number)
+                    {
+                        // called from MyImport aka UnmanagedFunc
+                        Console.WriteLine($"MyExport({number}) -> 42");
+                        return 42;
+                    }
+
+                    [DllImport("local", EntryPoint = "UnmanagedFunc")]
+                    public static extern void UnmanagedFunc(); // calls ManagedFunc aka MyExport
+                }
+                """;
+            string cCode =
+                    """
+                    #include <stdio.h>
+
+                    int ManagedFunc(int number);
+
+                    void UnmanagedFunc()
```

Comment:
I guess this is where you insert some Chinese characters, like `鿀蜒枛遫䡫`

Reply:
Issue is more about these characters https://github.com/dotnet/runtime/blob/3eb8c7f1086b79b28a27b57a935f97be3b7fcccb/src/tasks/WasmAppBuilder/ManagedToNativeGenerator.cs#L39  and in this test case it will have `< >` special characters.

Reply:
I'm just saying that some Unicode characters are multi-byte and it should be tested.

Reply:
Thanks, will add test case for that.